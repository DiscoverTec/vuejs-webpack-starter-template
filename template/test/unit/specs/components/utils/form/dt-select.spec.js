import VeeValidate from 'vee-validate'
import { shallow, createLocalVue } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import Component from '@/components/utils/form/dt-select'

const localVue = createLocalVue()

localVue.use(VeeValidate)

describe(`Component ${Component.name}`, () => {

  it('should render default properties', () => {
    const renderer = createRenderer()
    const wrapper = shallow(Component, {
      localVue,
      propsData: {
        name: 'testinput',
        label: 'Test Input'
      }
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('should render with all properties', () => {
    const renderer = createRenderer()
    const wrapper = shallow(Component, {
      localVue,
      propsData: {
        value: '1',
        name: 'testinput',
        label: 'Test Input',
        validate: 'required',
        options: [
          { name: 'one', value: '1' },
          { name: 'two', value: '2' },
          { name: 'three', value: '3' }
        ]
      }
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('should render error message', async () => {
    const renderer = createRenderer()
    const wrapper = shallow(Component, {
      localVue,
      propsData: {
        name: 'testinput',
        label: 'Test Input',
        validate: 'required'
      }
    })
    await wrapper.vm.$validator.validateAll()
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('should display disabled select with options', () => {
    const renderer = createRenderer()
    const wrapper = shallow(Component, {
      localVue,
      propsData: {
        value: 1,
        disabled: true,
        name: 'testinput',
        label: 'Test Input',
        validate: 'required',
        options: [
          { name: 'one', value: '1' },
          { name: 'two', value: '2' },
          { name: 'three', value: '3' }
        ]
      }
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('should display disabled select without options', () => {
    const renderer = createRenderer()
    const wrapper = shallow(Component, {
      localVue,
      propsData: {
        value: '1',
        disabled: true,
        name: 'testinput',
        label: 'Test Input'
      }
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})