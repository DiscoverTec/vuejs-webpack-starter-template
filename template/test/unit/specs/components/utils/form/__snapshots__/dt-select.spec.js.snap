// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component dt-select should display disabled select with options 1`] = `
<div class="form-group">
    <label class="control-label"> Test Input: </label>
    <input placeholder="Test Input" disabled="disabled" value="one" class="form-control">
</div>
`;

exports[`Component dt-select should display disabled select without options 1`] = `
<div class="form-group">
    <label class="control-label"> Test Input: </label>
    <input placeholder="Test Input" disabled="disabled" value="1" class="form-control">
</div>
`;

exports[`Component dt-select should render default properties 1`] = `
<div class="form-group">
    <label class="control-label"> Test Input: </label>
    <select name="testinput" class="form-control"></select>
    <small class="text-danger" style="display:none;">

    </small>
</div>
`;

exports[`Component dt-select should render error message 1`] = `
<div class="form-group has-warning">
    <label class="control-label"> Test Input: </label>
    <select name="testinput" class="form-control"></select>
    <small class="text-danger">
        The testinput field is required.
    </small>
</div>
`;

exports[`Component dt-select should render with all properties 1`] = `
<div class="form-group">
    <label class="control-label"> Test Input: </label>
    <select name="testinput" class="form-control">
        <option selected="selected" value="1">
            one
        </option>
        <option value="2">
            two
        </option>
        <option value="3">
            three
        </option>
    </select>
    <small class="text-danger" style="display:none;">

    </small>
</div>
`;
